### PR TITLE
grex: 1.4.2 -> 1.4.4

### PR DESCRIPTION
**Pull request by mfrw**
- grex: add mfrw as maintainer
- grex: 1.4.2 -> 1.4.4

## Description of changes
- Changelog: https://github.com/pemistahl/grex/releases/tag/v1.4.4
- Diff: https://github.com/pemistahl/grex/compare/v1.4.2...v1.4.4

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mfrw** [comment]
Result of `nixpkgs-review pr 251182` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>grex</li>
  </ul>
</details>

File: pkgs/tools/misc/grex/default.nix [hunk 1]
```diff
@@ -6,16 +6,16 @@
...
   pname = "grex";
-  version = "1.4.2";
+  version = "1.4.4";
 
   src = fetchFromGitHub {
     owner = "pemistahl";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-kbrBnpoVeXSgmIqElpfAnoSutYwsnFaxFJpRsJPUE8k=";
+    sha256 = "sha256-ef1eUxeCznIgXLoywwJmnLkTGdW1AmGwCin9DLU9kAs=";
```
Comment (AndersonTorres) [inline pkgs/tools/misc/grex/default.nix:15]:
```suggestion
    hash = "sha256-ef1eUxeCznIgXLoywwJmnLkTGdW1AmGwCin9DLU9kAs=";
```

File: pkgs/tools/misc/grex/default.nix [hunk 2]
```diff
@@ -6,16 +6,16 @@
...
   };
 
-  cargoSha256 = "sha256-WO8h5JUEiCpfsROQoRZj7I6em6REcJ2PfmoGN1zKRPg=";
+  cargoSha256 = "sha256-XLH+fS3fwRcWmVOzTjUacV010N37Oofs9Tbixdka1qY=";
```
Comment (AndersonTorres) [inline pkgs/tools/misc/grex/default.nix:18]:
```suggestion
  cargoHash = "sha256-XLH+fS3fwRcWmVOzTjUacV010N37Oofs9Tbixdka1qY=";
```

File: pkgs/tools/misc/grex/default.nix [hunk 3]
```diff
@@ -28,6 +28,6 @@ rustPlatform.buildRustPackage rec {
     description = "A command-line tool for generating regular expressions from user-provided test cases";
     homepage = "https://github.com/pemistahl/grex";
     license = licenses.asl20;
-    maintainers = with maintainers; [ SuperSandro2000 ];
+    maintainers = with maintainers; [ SuperSandro2000 mfrw ];
   };
```
Comment (AndersonTorres) [inline pkgs/tools/misc/grex/default.nix:32]: Please include `mainProgram` and `changelog`

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/grex/default.nix (hunk 3 above)
Comment (mfrw) [reply pkgs/tools/misc/grex/default.nix:34]: Sure :)

File: pkgs/tools/misc/grex/default.nix [hunk 4]
```diff
@@ -27,7 +27,9 @@ rustPlatform.buildRustPackage rec {
   meta = with lib; {
     description = "A command-line tool for generating regular expressions from user-provided test cases";
     homepage = "https://github.com/pemistahl/grex";
+    changelog = "https://github.com/pemistahl/grex/releases/tag/v${version}";
     license = licenses.asl20;
-    maintainers = with maintainers; [ SuperSandro2000 ];
+    mainProgram = "weggli";
```
Comment (mfrw) [inline pkgs/tools/misc/grex/default.nix:32]:
```suggestion
    mainProgram = "grex";
```

File: pkgs/tools/misc/grex/default.nix [hunk 5]
```diff
@@ -6,16 +6,16 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "grex";
-  version = "1.4.2";
+  version = "1.4.4";
 
   src = fetchFromGitHub {
     owner = "pemistahl";
     repo = pname;
```
Comment (AndersonTorres) [inline pkgs/tools/misc/grex/default.nix:13]:
```suggestion
    repo = "grex";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/grex/default.nix (hunk 5 above)
Comment (mfrw) [reply pkgs/tools/misc/grex/default.nix]: Done.

**Review by AndersonTorres — APPROVED** [review]

**Review by natsukium — APPROVED** [review]
The diff looks good to me.

**Pull request merged**